airflow/providers/google/marketing_platform/operators/search_ads.py

```diff
@@ -19,217 +19,278 @@
-        Service Account Token Creator IAM role to the directly preceding identity, with first
-        account from the list granting this role to the originating account (templated).
+    :param api_version: The version of the API that will be requested for example 'v0'.
     """
 
     template_fields: Sequence[str] = (
-        "report",
-        "impersonation_chain",
+        *_GoogleSearchAdsBaseOperator.template_fields,
+        "page_token",
+        "page_size",
     )
-    template_ext: Sequence[str] = (".json",)
 
     def __init__(
         self,
         *,
-        report: dict[str, Any],
-        api_version: str = "v2",
-        gcp_conn_id: str = "google_cloud_default",
-        delegate_to: str | None = None,
-        impersonation_chain: str | Sequence[str] | None = None,
+        customer_id: str,
+        query: str,
+        page_token: str | None = None,
+        page_size: int = 10000,
+        return_total_results_count: bool = False,
+        summary_row_setting: str | None = None,
+        validate_only: bool = False,
         **kwargs,
     ) -> None:
         super().__init__(**kwargs)
-        self.report = report
-        self.api_version = api_version
-        self.gcp_conn_id = gcp_conn_id
-        self.delegate_to = delegate_to
-        self.impersonation_chain = impersonation_chain
-
-    def prepare_template(self) -> None:
-        # If .json is passed then we have to read the file
-        if isinstance(self.report, str) and self.report.endswith(".json"):
-            with open(self.report) as file:
-                self.report = json.load(file)
+        self.customer_id = customer_id
+        self.query = query
+        self.page_token = page_token
+        self.page_size = page_size
+        self.return_total_results_count = return_total_results_count
+        self.summary_row_setting = summary_row_setting
+        self.validate_only = validate_only
 
     def execute(self, context: Context):
-        hook = GoogleSearchAdsHook(
-            gcp_conn_id=self.gcp_conn_id,
-            delegate_to=self.delegate_to,
-            api_version=self.api_version,
-            impersonation_chain=self.impersonation_chain,
+        self.log.info("Querying Search Ads")
+        response = self.hook.search(
+            customer_id=self.customer_id,
+            query=self.query,
+            page_size=self.page_size,
+            page_token=self.page_token,
+            return_total_results_count=self.return_total_results_count,
+            summary_row_setting=self.summary_row_setting,
+            validate_only=self.validate_only,
         )
-        self.log.info("Generating Search Ads report")
-        response = hook.insert_report(report=self.report)
-        report_id = response.get("id")
-        self.xcom_push(context, key="report_id", value=report_id)
-        self.log.info("Report generated, id: %s", report_id)
+        self.log.info("Query result: %s", response)
         return response
```

Comment:
@molcay sorry for mentioned it here, because we needed to discuss it earlier in DesDoc. But I have noticed that each operator returns pure HTTP Response. It shouldn't be like this. Each operator should return some data or return nothing(for delete operation for example). In your case you need to parse each response in the execute method and then return data which should сorrelate with the Operator name. For example 
GoogleSearchAdsSearchOperator -> returns Report obj in dict format; GoogleSearchAdsGetFieldOperator -> returns Field obj in dict format; GoogleSearchAdsListCustomColumnsOperator -> returns List obj in dict format; and etc.

This is important, because in `execute` method return value is sent to XCom and with pure HTTP Response it can bring to an errors.

Reply:
Ohh, I was thinking that this `response` is already a dictionary. I will do the changes accordingly then.